include/seqan3/io/sam_file/detail/cigar.hpp

```diff
@@ -266,7 +266,7 @@ template <seqan3::detail::pairwise_alignment alignment_type>
 [[nodiscard]] inline std::string get_cigar_string(std::vector<cigar> const & cigar_vector)
 {
     std::string result{};
-    std::ranges::for_each(cigar_vector, [&result] (auto & cig) { result.append(cig.to_string()); });
+    std::ranges::for_each(cigar_vector, [&result] (auto & cig) { std::string s = cig.to_string(); result.append(s); });
```

Comment:
This results in an extra dynamic allocation.

This avoids it:
```suggestion
    std::ranges::for_each(cigar_vector, [&result] (auto & cig) { result.append(static_cast<std::string_view>(s.to_string())); });
```

Reply:
I explicitly avoided this cast. Notably, without any change (the original:)
```cpp
std::ranges::for_each(cigar_vector, [&result] (auto & cig) { result.append(s.to_string()); });
```
The `to_string()` returns a `small_string` that is implicitly converted to a `string_view`.

So it's the same as your suggestion.

The problem is that this causes the following to fail:
```cpp
        auto seq_ref = "ACGTGATCTG"_dna4;
        auto seq_read = "ACGTCGTAGTG"_dna4;
        std::vector<seqan3::gapped<seqan3::dna4> ref;
        std::vector<seqan3::gapped<seqan3::dna4> read;
        TestFixture::initialise_typed_test_container(ref,  seq_ref);
        TestFixture::initialise_typed_test_container(read, seq_read);
        seqan3::insert_gap(ref, std::ranges::next(begin(ref), 7), 2);
        seqan3::insert_gap(read, std::ranges::next(begin(read), 4), 1);

        std::string expected = "4M1D2M2I3M";

        EXPECT_EQ(expected, seqan3::detail::get_cigar_string(std::make_pair(ref, read)));
```
with error:
```
  expected
    Which is: "4M1D2M2I3M"
  seqan3::detail::get_cigar_string(std::make_pair(ref, read))
    Which is: "4M\0\0\0\0\0\0\0\0\01D\0\0\0\0\0\0\0\0\02M\0\0\0\0\0\0\0\0\02I\0\0\0\0\0\0\0\0\03M\0\0\0\0\0\0\0\0\0"
```

I thought this happens because the `string_view` is parsed incorrectly by accessing the underlying c string as the range to parse which makes the end incorrect.

Reply:
The cast to string_view was incorrect, see my other comment. If you change that, it works.

I would also explicitly cast to string_view, since append() also takes a string and that overload is unconstrained. Also the string_view overload is explicitly disabled if the type is also convertible to `char *` which `small_string` is.
I would actually suggest making small_string's conversion operator to string explicit, because it allocates, but that's a different question.

Reply:
Thanks!